### PR TITLE
update docs, add crossvalidation to evaluation script

### PR DESCRIPTION
**Pull request by leachim**
**Proposed changes**:
- ...

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog

apologies for the inconvenience. We can delete the other request.

### CODE REVIEW

File: rasa_nlu/evaluate.py [hunk 1]
```diff
@@ -24,17 +25,22 @@
...
-            description='evaluate a trained Rasa NLU pipeline')
+            description='evaluate a Rasa NLU pipeline with cross validation or on external data')
 
-    parser.add_argument('-d', '--data', default=None,
-                        help="file containing evaluation data")
-    parser.add_argument('-m', '--model', required=True,
-                        help="path to model")
+    parser.add_argument('-d', '--data', required=True,
+                        help="file containing training/evaluation data")
+    parser.add_argument('--mode', required=True,
```
Comment (tmbo) [inline rasa_nlu/evaluate.py]: let's make the `evaluation` mode the default mode (for backwards compatibility) and make this flag optional (as well as remove the flag from the evaluation example)

**Comment by leachim** [comment]
updated

**Comment by tmbo** [comment]
Can you also please write a test for the two functions (not the main code)

File: tests/evaluation.py [hunk 2]
```diff
@@ -185,6 +186,32 @@ def test_duckling_patching():
...
+
+    np.seed(2018)
+    results = run_cv_evaluation(td, n_folds, nlu_config)
+
+    rel_tol=1e-09
+    abs_tol=0.01
+
+    a = np.mean(results["accuracy"])
+    b = 0.65 # expected result
+    assert(abs(a-b) <= max(rel_tol * max(abs(a), abs(b)), abs_tol))
```
Comment (tmbo) [inline tests/evaluation.py]: can we use numpy here instead? https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.testing.assert_approx_equal.html

**Comment by leachim** [comment]
I wasn't too sure about testing the run_cv_evaluation method. Any change to spacy for example could break it. What do you think?

**Comment by tmbo** [comment]
No thats fine, tests are supposed to be run with a specific model. So yes, they might break with a different models but that is ok.

**Comment by tmbo** [comment]
Ok looking good :+1: @leachim can you please add the change to the `CHANGELOG.rst`, other than that I think we are good to go.

**Comment by amn41** [comment]
@tmbo @leachim I have one more request :D can we please import matplotlib only where it's needed? e.g. inside the `plot_confusion_matrix` function?

**Comment by leachim** [comment]
Sorry, had to rebase it again. Hope it's ok now.

**Comment by tmbo** [comment]
looks good, nice work :+1:

**Pull request merged**